canopen/objectdictionary/__init__.py

```diff
@@ -285,6 +285,9 @@ def decode_raw(self, data):
             return data.decode("utf_16_le")
         elif self.data_type in self.STRUCT_TYPES:
             try:
+                needed_bytes = self.STRUCT_TYPES[self.data_type].size
+                if needed_bytes < len(data):
+                    data = bytearray(data)[0:needed_bytes]
```

Comment:
Does it need to be a `bytearray`?

Reply:
Not quite sure why I couldn't use `data[0:needed_bytes]`, but I think it had something to do with mutable types?

Anyway, `encode_raw()` uses `bytearray`, and so does [python-can](https://python-can.readthedocs.io/en/stable/message.html#can.Message.data).  What other type did you have in mind?

Reply:
The data argument will be either bytes or bytearray which are both supported by the struct module so it should work. It doesn’t have to be mutable.

Reply:
Okay, removed the conversion and now slicing `data` directly.